src/Microsoft.AspNetCore.Server.HttpSys/MessagePump.cs

```diff
@@ -32,7 +33,7 @@ internal class MessagePump : IServer
 
         private readonly ServerAddressesFeature _serverAddresses;
 
-        public MessagePump(IOptions<HttpSysOptions> options, ILoggerFactory loggerFactory)
```

Comment:
Updated to resolve IAuthenticationSchemeProvider as enumerable and only throw if AuthentionSchemes for the server options is not none.